Week4/Forecast.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import ForecastTab from './ForecastTab';
+import { Image } from 'react-native';
+
+export type StackParamList = {
+    ForecastTab5: { days: number };
+    ForecastTab7: { days: number };
+};
+
+const Tab = createBottomTabNavigator<StackParamList>();
+
+const ForecastScreen: React.FC = () => {
+    return (
```

Comment:
I would avoid placing a navigator inside of your screen. I'd place this Tab Navigator in `Week4.tsx`.

So you'd have a Week4Drawer navigator and a Week4Tab navigator component in there. Then you'd end up passing the Week4Tab navigator to the component for your forecast screen inside your Week4Drawer.

It would look something like this, which would allow you to get rid of `Forecast.tsx` too
```
const Week4 = () => {
  return (
    <Drawer.Navigator initialRouteName="CurrentWeather">
      <Drawer.Screen name="CurrentWeather" component={CurrentWeather} />
      <Drawer.Screen name="Forecast" component={Week4Tab} />
    </Drawer.Navigator>
  );
};

export const Week4Tab = () => {
  <Tab.Navigator>
    <Tab.Screen
      name="ForecastTab5"
      component={ForecastTab}
      initialParams={{ days: 5 }}
      options={{
        headerShown: false,
        tabBarLabel: 'Five day',
        tabBarIcon: ({ color, size }) => (
          <Image
            source={{
              uri: 'https://upload.wikimedia.org/wikipedia/commons/4/4e/5NumberFiveInCircle.png',
            }}
            style={{ width: size, height: size, tintColor: color }}
          />
        ),
      }}
    />
    <Tab.Screen
      name="ForecastTab7"
      component={ForecastTab}
      initialParams={{ days: 7 }}
      options={{
        headerShown: false,
        tabBarLabel: 'Seven Day',
        tabBarIcon: ({ color, size }) => (
          <Image
            source={{
              uri: 'https://upload.wikimedia.org/wikipedia/commons/d/d7/7NumberSevenInCircle.png',
            }}
            style={{ width: size, height: size, tintColor: color }}
          />
        ),
      }}
    />
  </Tab.Navigator>;
};
```